loader/loader.go

```diff
@@ -0,0 +1,281 @@
+package loader
+
+import (
+	"context"
+	"fmt"
+	ecapiv1alpha1 "github.com/hacbs-contract/enterprise-contract-controller/api/v1alpha1"
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"github.com/redhat-appstudio/release-service/api/v1alpha1"
+	"github.com/redhat-appstudio/release-service/tekton"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// convertUnstructuredObject converts an unstructured object into an instance of the crd passed as an argument.
+func convertUnstructuredObject[T any](unstructuredObject *unstructured.Unstructured, crd *T) error {
+	return runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredObject.UnstructuredContent(), crd)
+}
+
+// getObject loads an object from the cluster. This is a generic function that requires the object to be passed as an
+// argument. The object is modified during the invocation, but it's returned as well to simplify the implementation of
+// functions using this function.
+func getObject[T any](name, namespace string, cli client.Client, ctx context.Context, object *T) (*T, error) {
```

Comment:
I think this makes more sense in the goodies repo

Reply:
To be done once the loader is established in the release-service and in the internal-services repos.